streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -745,9 +780,21 @@ protected void maybeCommit(final long now) {
      * Commit the states of all its tasks
      */
     private void commitAll() {
-        for (final StreamTask task : activeTasks.values()) {
-            commitOne(task);
+        final RuntimeException e = performOnStreamTasks(new StreamTaskAction() {
+            @Override
+            public String name() {
+                return "commit";
+            }
+
+            @Override
+            public void apply(final StreamTask task) {
+                commitOne(task);
+            }
+        });
+        if (e != null) {
+            throw e;
```

Comment:
Do we always want to rethrow all other exceptions other than `producerFenced`? For example `consumer#commit` could return `CommitFailedException` here, which means:

```
if the commit failed and cannot be retried.
     *             This can only occur if you are using automatic group management with {@link #subscribe(Collection)},
     *             or if there is an active group with the same groupId which is using group management.
```

In this case it means a rebalance has likely happened and hence the consumer no longer owns the partition that it tries to commit to. In this case shouldn't we handle it internally than throwing it all the way to the user?